### PR TITLE
feat: Add hat.Header for adding headers to a request

### PR DESCRIPTION
**Pull request by Emyrk**
Do I need to update the README LoC and APISymbols?

Is there a script? Or did someone hand count those?

### CODE REVIEW

File: request_test.go [hunk 1]
```diff
@@ -1,20 +1,37 @@
-package hat
+package hat_test
```
Comment (jawnsy) [inline request_test.go:1]: nice

File: request_test.go [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
 	"net/url"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"go.coder.com/hat"
+	"go.coder.com/hat/asshat"
 )
 
 func TestURLParams(t *testing.T) {
```
Comment (jawnsy) [inline request_test.go:15]: totally meaningless change but can we add t.Parallel()?

File: request_test.go [hunk 3]
```diff
@@ -1,20 +1,37 @@
...
-	URLParams(url.Values{
+	hat.URLParams(url.Values{
 		"q": []string{"sean"},
 	})(t, req)
 
 	require.Equal(t, "http://google.com?q=sean", req.URL.String())
 }
+
+func TestHeader(tt *testing.T) {
+	hKey, hVal := "X-Custom-Header", "test-value"
```
Comment (jawnsy) [inline request_test.go:27]: thoughts on making these const?

**Review by jawnsy — APPROVED** [review]
LGTM, I left some totally nitpicky comments, feel free to ignore

File: request_test.go (hunk 2 above)
Comment (Emyrk) [reply request_test.go:15]: sure

**Pull request merged**